muacrypt/recommendation.py

```diff
@@ -6,8 +6,34 @@ def __init__(self, account, peerstates):
         self.peerstates = peerstates
 
     def ui_recommendation(self):
-        return 'available'
+        # only consider first peer for now
+        peer = list(self.peerstates.values())[0]
+        return self._peer_recommendation(peer)
 
     def target_keys(self):
-        return {addr: state.public_keyhandle for addr, state in
+        return {addr: self._target_key(peer) for addr, peer in
                 self.peerstates.items()}
+
+    def _peer_recommendation(self, peer):
+        if peer is None:
+            return 'disable'
+        if len(peer.public_keyhandle):
+            return 'available'
+        else:
+            return 'disable'
+
+    def _target_key(self, peer):
+        return self._public_key(peer) or self._gossip_key(peer)
+
+    def _public_key(self, peer):
+        if peer:
+            return self._key(peer.public_keyhandle)
+
+    def _gossip_key(self, peer):
+        # gossip keyhandle is not implemented yet.
+        if peer and hasattr(peer, 'gossip_keyhandle'):
+            return self._key(peer.gossip_keyhandle)
+
+    def _key(self, handle):
+        if len(handle):
```

Comment:
``Recommendation`` is not using ``self.account`` at the moment.  So unless you have a direct use of  ``account`` in this PR i suggest to remove it from the ``init`` and also from the tests.  The spec also does not indicate that there is a dependency on "own state" and even if so, we should pass in ``OwnState`` or those attributes needed, and not the whole ``account`` object which is rather large.

Reply:
Ahh... i see. I think the only thing i will need from account is my own prefer_encrypt. And maybe if i have a key myself. But you are right. Can add that later.

Reply:
Just for the record -- when working test-driven it's good to only add those attributes which are needed.  Particularly ``account`` is a "large" object with a lot of behaviour.  Being more explicit and minimal about what functions/helpers needed keeps them more readable and easier to reason about.